[AIG] Updated logging (#16793)
cloudflare/cloudflare-docs: src/content/docs/ai-gateway/observability/persistent-logs.mdx

---
pcx_content_type: reference
title: Persistent logs(beta)
sidebar:
  badge:
    text: Beta
---

import { Render } from "~/components";

> Logging is a fundamental building block for application development — it provides insights during the early stages of development and is often critical to understanding issues occurring in production.

Reviewer: ```suggestion
Logging is a fundamental building block for application development. Logging provides insights during the early stages of development and is often critical to understanding issues occurring in production.
```

Your AI Gateway dashboard shows logs of individual requests, including the user prompt, model response, provider, timestamp, request status, token usage, cost, and duration. These logs persist, giving you the flexibility to store them for your preferred duration and do more with valuable request data. You can store up to 10 million logs per gateway.

When you are nearing your plan limit, an alert will be displayed in the dashboard. If your limit is reached, new logs will stop being saved. To continue saving logs, you must delete older logs to free up space for new logs.

> To learn more about your plan limits, visit our [Pricing](/ai-gateway/pricing/) page.

Reviewer: ```suggestion
To learn more about your plan limits, refer to the [Pricing](/ai-gateway/pricing/) page.
```

Reviewer: We usually prefer to use "Refer to" instead of "Visit".

Author: Thanks for this.
